hm_1/tasks/task_01_05.py

```diff
@@ -0,0 +1,17 @@
+"""
+Given a list of integers numbers "nums".
+You need to find a sub-array with length less equal to "k", with maximal sum.
+The written function should return the sum of this sub-array.
+Examples:
+    nums = [1, 3, -1, -3, 5, 3, 6, 7], k = 3
+    result = 16
+"""
+from typing import List
+
+
+def find_maximal_subarray_sum(nums: List[int], k: int) -> int:
+    sums = []
+    for j in range(k, -1, -1):
+        for i in range(len(nums) - (j - 1)):
+            sums.append(sum(nums[i : i + j]))
+    return "Size of subarray > length of your list" if k > len(nums) else max(sums)
```

Comment:
Does the subarray of size 1 is always less or equal to k for any k?

Reply:
I think yes, in that case the function must give that answer - find_maximal_subarray_sum([4], 0) == 0

Reply:
I mean that you should check all possible subarray that has size <= k

`find_maximal_subarray_sum([4], 10) == 4`